src/openms/include/OpenMS/KERNEL/MassTrace.h

```diff
@@ -321,6 +321,11 @@ namespace OpenMS
     /// median of trace intensities
     double computeMedianIntensity_() const;
 
+    /// calculate x coordinate of start/end indexes at half_max
+    /// calculation is based on (yB - yA) / (xB - xA) = (yC - yA) / (xC - xA)
+    /// solve for xC: xC = xA + ((yC - yA) * (xB - xA) / (yB - yA))
+    double calculateXCoordinateAtHalfMax_(double xA, double xB, double yA, double yB, double fwhm_int) const;
```

Comment:
I suggest to change this to `linearInterpolationAtY_` and change `fwhm_int` to `y_eval` or `y` or something to make it more clear.

Reply:
I'll try to fix the tests. 

Yeah, renamed that while debugging, will revert the name to something reflecting the calculation again.